### PR TITLE
Scheduler files cleanup

### PR DESCRIPTION
**Pull request by csarofeen**
Break schedulers out into their own files. Put all scheduling logic under codgen/cuda/scheduler/

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/scheduler/normalization.h [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+TORCH_CUDA_API void scheduleNormalization(
+    Fusion* fusion,
+    const ReductionParams& rparams,
+    const std::vector<TensorView*>& reduction_tv,
+    std::vector<TensorView*>& other_tv);
+
+} // namespace cuda
+} // namespace fuser
+} // namespace jit
+} // namespace torch
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/normalization.h:31]: nit: extra line

File: torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp [hunk 2]
```diff
@@ -0,0 +1,657 @@
...
+    if (input->getRootDomain().size() != kReductionRootDims) {
+      scheduler_utils::handleCastBroadcastInput(fusion, input);
+    }
+  }
+}
+
+} // namespace cuda
+} // namespace fuser
+} // namespace jit
+} // namespace torch
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/normalization.cpp]: nit: extra line

File: torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp [hunk 3]
```diff
@@ -0,0 +1,543 @@
...
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+namespace {
+constexpr int kLoopUnrollSplit = 4;
+
+// Largest Power of 2 less-than n
+constexpr int lastPow2(int n) {
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp:17]: Maybe move to schedule_utils

File: torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+
+#include <ATen/cuda/CUDAContext.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+namespace {
+constexpr int kLoopUnrollSplit = 4;
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp]: should this be in the scheduleReduction function?

File: torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    out_tv->axis(2)->parallelize(ParallelType::TIDx);
+  }
+
+  return true;
+}
+
+} // namespace cuda
+} // namespace fuser
+} // namespace jit
+} // namespace torch
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/pointwise.cpp:72]: nit: extra line

File: torch/csrc/jit/codegen/cuda/scheduler/pointwise.h [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+TORCH_CUDA_CU_API bool scheduleFusion(
+    Fusion* fusion,
+    const at::ArrayRef<c10::IValue> inputs);
+
+TORCH_CUDA_CU_API bool scheduleFusion(Fusion* fusion);
+
+} // namespace cuda
+} // namespace fuser
+} // namespace jit
+} // namespace torch
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/pointwise.h:22]: nit: extra line

File: torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp [hunk 7]
```diff
@@ -0,0 +1,543 @@
...
+        }
+      }
+    }
+  }
+}
+
+} // namespace cuda
+} // namespace fuser
+} // namespace jit
+} // namespace torch
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp:543]: nit: extra line

File: torch/csrc/jit/codegen/cuda/scheduler/reduction.h [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+TORCH_CUDA_CU_API void scheduleReduction(
+    Fusion* fusion,
+    const ReductionParams& rparams,
+    TensorView* red_tv,
+    std::vector<TensorView*> outs_of_red);
+} // namespace cuda
+} // namespace fuser
+} // namespace jit
+} // namespace torch
```
Comment (rdspring1) [inline torch/csrc/jit/codegen/cuda/scheduler/reduction.h:31]: nit: extra line

File: torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp (hunk 3 above)
Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/scheduler/reduction.cpp:17]: I had it in there but it was complaining about a linking issue I couldn't figure out, so moved it back. Will consider in later refactors to see if I can get it sorted out.

**Review by rdspring1 — APPROVED** [review]
LGTM

**Pull request merged**